db/migrate/20160321160404_add_middleware_event_streams.rb

```diff
@@ -0,0 +1,7 @@
+class AddMiddlewareEventStreams < ActiveRecord::Migration[5.0]
+  def change
+    add_column :event_streams, :middleware_type, :string   # relevant type (entity class name)
+    add_column :event_streams, :middleware_id, :bigint     # MIQ entity id
```

Comment:
Incidentally, the preferred style *would be* `add_reference :event_streams, :middleware, :polymorphic => true, :type => :bigint`